src/server/api/routes/bookstore/index.js

```diff
@@ -1,121 +1,41 @@
-            property: 'Visible',
-            checkbox: {
-              equals: true,
-            },
-          },
-        ],
-      },
-      sorts: [
-        {
-          property: 'Listing Date',
-          direction: 'descending',
-        },
-      ],
-    });
+      res.set(
+        'Cache-Control',
+        'public, max-age=3600, stale-if-error=604800, stale-while-revalidate=604800'
+      );
 
-    const results = response.results.map(({ properties }) => ({
-      classId: parseClassId(properties['NFT Class ID'].rich_text[0].plain_text),
-      locales: properties.Locale.select
-        ? [properties.Locale.select.name]
-        : (properties.Locale.multi_select || []).map(({ name }) => name),
-      date: properties['Listing Date'].date.start,
-    }));
+      res.json(result);
+    } catch (error) {
+      handleRestfulError(req, res, next, error);
+    }
+  };
+}
 
-    res.set(
-      'Cache-Control',
-      'public, max-age=3600, stale-if-error=604800, stale-while-revalidate=604800'
-    );
-    res.json({ results });
-  } catch (error) {
-    handleRestfulError(req, res, next, error);
-  }
+Object.entries({
+  landing: 'Landing',
+  listing: 'Listing',
+  all: 'All',
+}).forEach(([path, view]) => {
+  router.get(`/bookstore/lists/${path}`, getBookstoreListRequestHandler(view));
 });
 
-router.get('/bookstore/lists', async (req, res, next) => {
-  try {
-    const response = await notion.databases.query({
-      database_id: NOTION_DATABASE_ID_FOR_BOOKSTORE_LISTS,
-    });
-
-    const results = {};
-    response.results.forEach(({ properties }) => {
-      const listId = properties.ID.title[0].plain_text;
-      const classIdPropList = properties['Class ID List'].rollup.array;
-      const titlePropList = properties['Name List'].rollup.array;
-      const localePropList = properties['Locale List'].rollup.array;
-      const visibilityPropList = properties['Visibility List'].rollup.array;
-      const drmFreeList = properties['DRM-free List'].rollup.array;
-
-      results[listId] = [];
-      classIdPropList.forEach((classId, index) => {
-        const isVisible = visibilityPropList[index].formula.boolean;
-        if (!isVisible) return;
-
-        const isDRMFree = drmFreeList[index].checkbox;
-        const localeProp = localePropList[index];
-        results[listId].push({
-          classId: parseClassId(classId.rich_text[0].plain_text),
-          title: (titlePropList[index].title[0] || {}).plain_text || '',
-          isDRMFree,
-          locales: localeProp.select
-            ? [localeProp.select.name]
-            : (localeProp.multi_select || []).map(({ name }) => name),
-        });
-      });
-    });
-
-    res.set(
-      'Cache-Control',
-      'public, max-age=3600, stale-if-error=604800, stale-while-revalidate=604800'
-    );
-    res.json({ results });
-  } catch (error) {
-    handleRestfulError(req, res, next, error);
-  }
-});
+router.get('/bookstore/lists/:id', getBookstoreListRequestHandler());
```

Comment:
Would it be more consistent if we pass `req.params.id` and `view` respectively to a same interface? i.e. `getBookstoreListRequestHandler(view)`  `getBookstoreListRequestHandler(req.params.id)`